src/serial.rs

```diff
@@ -101,7 +140,13 @@ macro_rules! hal {
         ),
     )+) => {
         $(
+            $(#[$meta])*
             impl<PINS> Serial<$USARTX, PINS> {
+
+            /// Configures the Serial interface and creates an interface struct.
+            /// `baud_rate` configures the baud rate of the interface. The other parameters are
+            /// handles to the corresponding registers, to make Rust's borrow checker happy.
```

Comment:
The regs aren't passed to make the borrow checker happy, they are passed to be modified in order to make the device work.

Reply:
Yes, but without the borrow checker, the registers could be encoded in the function statically (C-style), or have I misunderstood the concept of the peripheral struct?

I thought, that this is the reason that 6 parameters are passed to that function, whilst only one is a real parameter.

Reply:
(But yes, that is sloppy language)